lib/spack/spack/environment.py

```diff
@@ -617,7 +694,88 @@ def install(self, user_spec, concrete_spec=None, **install_args):
                 concrete = spec.concretized()
                 self._add_concrete_spec(spec, concrete)
 
-        concrete.package.do_install(**install_args)
+        self._install(concrete, **install_args)
+
+    def _install(self, spec, **install_args):
+        # Make sure log directory exists
+        log_path = self.log_path
+        fs.mkdirp(log_path)
+
+        with fs.working_dir(self.path):
+            spec.package.do_install(**install_args)
```

Comment:
does the working directory need to be changed during the `do_install` call or during the linking phase below?  If it's not necessary to change the working dir during the actual build, then don't (I realize it probably shouldn't matter, but it seems like an unneeded change to include this in the `with`)